### PR TITLE
Added ERC223 feature

### PR DESCRIPTION
**Pull request by ProphetDaniel**

### CODE REVIEW

File: contracts/token/ERC223/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  4. Ether transactions and token transactions behave different: one of the goals of developing ERC223 was to make token transactions similar to Ether transactions to avoid users mistakes when transferring tokens and make interaction with token transactions easier for contract developers.
+  
+### ERC223 advantages.
+  1. Provides a possibility to avoid accidentally lost tokens inside contracts that are not designed to work with sent tokens.
+  2. Allows users to send their tokens anywhere with one function `transfer`. No difference between is the receiver a contract or not. No need to learn how token contract is working for regular user to send tokens.
+  3. Allows contract developers to handle incoming token transactions.
+  4. ERC223 `transfer` to contract consumes 2 times less gas than ERC20 `approve` and `transferFrom` at receiver contract.
+  5. Allows to deposit tokens intor contract with a single transaction. Prevents extra blockchain bloating.
```
Comment (DavidKnott) [inline contracts/token/ERC223/README.md]: `intor` to `into`

**Comment by DavidKnott** [comment]
@ProphetDaniel This looks like it'll be a great addition!  Could you explain how having 2 `transfer` functions in the same contract works? (I'm curious)  I'm assuming it's smart enough to choose which function to use based on if `bytes data` is provided?

**Comment by Dexaran** [comment]
@DavidKnott this transfer functions would have different signatures.
The signature of `transfer(address,uint256)` function is [0xa9059cbb](https://www.4byte.directory/signatures/?bytes4_signature=a9059cbb).
The signature of `transfer(address,uint256,bytes)` function is [0xbe45fd62](https://www.4byte.directory/signatures/?bytes4_signature=be45fd62).
You are right, if `bytes data` is provided, then the signature of the called function will match the `transfer (address, uint256, bytes)`, and if the data is not provided, the signature will match  the `transfer (address, uint256)`.

**Comment by ProphetDaniel** [comment]
Thank you for answering him @Dexaran!

**Comment by izqui** [comment]
We have a more complex implementation we did at https://github.com/aragon/erc23 that has the `tokenPayable`modifier that i think it is very interesting for using tokens as ether for calling functions

**Comment by DavidKnott** [comment]
@Dexaran @izqui  Thanks a lot for explaining this!

**Comment by ProphetDaniel** [comment]
@izqui , Indeed tokenPayable modifier is very interesting. Can you make a PR to this [feature branch](https://github.com/ProphetDaniel/zeppelin-solidity/tree/feature/ERC223)?

**Comment by ProphetDaniel** [comment]
Another point to be considered is the cost for the added complexity the tokenPayable modifyer adds. Anyway I think it a good practice to implement first the basic ERC223 token standard. And later on the more complex implementation that adds features like the tokenPayable modifyer.

**Comment by se3000** [comment]
@izqui I like your `tokenPayable` modifier and `tkn` storage technique. It adds a little complexity, but would be a good example for other contracts in the future.

It looks like your `tokenFallback` function accepts an extra parameter, `address _origin`, which is not specified in the original spec. Can you clarify for me why it is a worth while parameter to add?

**Comment by maraoz** [comment]
@ProphetDaniel this still needs rebasing to be mergeable.

**Comment by maraoz** [comment]
@ProphetDaniel still needs to be rebased correctly. Remember not to do a `git pull` before pusing, and do a `git push -f` instead

**Comment by frangio** [comment]
Closing as stale. Furthermore, our implementation shouldn't be a clone of the reference implementation, but written from scratch.

**Comment by kylerschin** [comment]
Please reopen this @frangio . ERC223 tokens are necessary for storing tokens in my contract. I am building my token around the ERC223 for payment solutions and anti fraud using smart contracts between 2 people. I think people would like a safe way to use tokens such as these ones.

**Comment by frangio** [comment]
Hi @kylerschin. We appreciate your input. This PR was closed for different reasons. Please refer to #346, we'd love to have your opinion over there.

**Comment by ProphetDaniel** [comment]
It is quite amusing this specific actitude @frangio.
>Closing as stale.

@kylerschin, I tested this implementation with truffle tests and LETH wallet. More tests can be added though. One of the important questions is if it will work flawlessly with ERC20 wallets that could be awaiting ERC20 events for example. I took special care with that for the event triggering strategy. A great real life token wallet with which I have also tested is Inzhoop LETH.

**Pull request closed without merging**